icu4c/source/common/ulocbuilder.cpp

```diff
@@ -134,17 +134,12 @@ int32_t ulocbld_buildLanguageTag(ULocaleBuilder* builder,
         return 0;
     }
     icu::Locale l = INTERNAL(builder)->build(*err);
-    if (U_FAILURE(*err)) { return 0; }
-    CheckedArrayByteSink sink(buffer, bufferCapacity);
-    l.toLanguageTag(sink, *err);
-    if (U_FAILURE(*err)) { return 0; }
-    int32_t reslen = sink.NumberOfBytesAppended();
-    if (sink.Overflowed()) {
-        *err = U_BUFFER_OVERFLOW_ERROR;
-    } else {
-        u_terminateChars(buffer, bufferCapacity, reslen, err);
-    }
-    return reslen;
+    return icu::ByteSinkUtil::viaByteSinkToTerminatedChars(
+        buffer, bufferCapacity,
+        [&](icu::ByteSink& sink, UErrorCode& status) {
+            l.toLanguageTag(sink, status);
+        },
```

Comment:
question: can this be simplified to just
```suggestion
        l.toLanguageTag(sink, status),
```
?

Reply:
Eh, no, how would that work!?

Reply:
it's already a function that takes a sink and a status and returns void

Reply:
Well, sure, it is, but the code you suggested would call the function and then pass its void return value as the lambda argument to `viaByteSinkToTerminatedChars()`, and besides that, it's a member function of a class so in order to call it the caller needs to somehow also be given some reference to the object upon which the function should be called (to use as the `this` pointer), which here is the reference `l` which gets passed to `viaByteSinkToTerminatedChars()` through the lambda capture.